### DISCUSSION TITLE
Trying to understand why `copybuffer` plugin is not working

### DISCUSSION THREAD

**Discussion by rivenirvana** [ohmyzsh/ohmyzsh#12379] [Help] [answered]
- Relevant system details: unmodified `which` package and `omz` install
![image](https://github.com/ohmyzsh/ohmyzsh/assets/43519644/c08b3f4b-f145-43c4-9c13-e9cdfb32330d)

- With the default packaged `which` alias, `clipcopy` is not found and `copybuffer` does not work:
![image](https://github.com/ohmyzsh/ohmyzsh/assets/43519644/1ad78ad5-5e09-4900-83bb-b3b4453eb73e)

- With an alternative `which` alias, using `--read-functions`, `clipcopy` is found but `copybuffer` does not work:
![image](https://github.com/ohmyzsh/ohmyzsh/assets/43519644/8149cd2e-44f2-4fd5-b973-a680d42fbc64)

- With `which` unaliased and shell built-in used, `clipcopy` is found but `copybuffer` still does not work:
![image](https://github.com/ohmyzsh/ohmyzsh/assets/43519644/b0f2c310-4f5d-43ee-ac0b-158074802e6e)

- With `which` unaliased and using the binary directly, `clipcopy` is not found and `copybuffer` also does not work:
![image](https://github.com/ohmyzsh/ohmyzsh/assets/43519644/ad62215a-74cc-48cc-acab-e4ac3374d64b)

**Comment by rivenirvana** [comment]
I have checked the source and what related issues there are, but can't seem to pin this down and make it work. What might I be missing?

**Comment by mcornella** [comment, marked as answer]
Fixed in latest commit to always use the `builtin` which. For zsh, you shouldn't need this anyway since `which` already shows aliases and functions.

**Comment by rivenirvana** [comment]
The aliased `which` is custom packaged and a protected dependency of `grub2` in Fedora, unfortunately so I can't just remove it. Thanks a lot for the fix!
